piff/optical_model.py

```diff
@@ -175,52 +268,211 @@ def fit(self, star):
 
         # compute chisq
         chisq = np.std(image.array - model_image.array)
-        dof = np.count_nonzero(weight.array)
+        dof = np.count_nonzero(weight.array) - 6
 
-        var = np.zeros(len(star.fit.params)) 
+        var = np.zeros(len(star.fit.params))
         fit = StarFit(star.fit.params, params_var=var, flux=star.fit.flux,
                       center=star.fit.center, chisq=chisq, dof=dof)
         return Star(star.data, fit)
 
-    def getProfile(self, params):
-        """Get a version of the model as a GalSim GSObject
+    @lru_cache(maxsize=8)
+    def getOptics(self, zernike_coeff):
+        """ Get the strictly Optics PSF component.  Use two phase screens, one from mirror_figure_im, and the other from
+        zernike aberrations
 
-        :param params:      A np array with [z4, z5, z6...z11]
+        : param zernike_coeff:          A ndarray or list with Zernike Coefficients, in units of waves. Array indexed as [0, 0, z2, z3, z4,...z11...]
 
-        :returns: a galsim.GSObject instance
+        : returns: a galsim.GSObject instance
```

Comment:
You have an extra space here.  And on `: param`.

Reply:
Fixed